lib/v2/kantarworldpanel/index.js

```diff
@@ -0,0 +1,94 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const { art } = require('@/utils/render');
+const path = require('path');
+
+module.exports = async (ctx) => {
+    const { region = 'cn-en', category = 'news' } = ctx.params;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 30;
+
+    const rootUrl = 'https://www.kantarworldpanel.com';
+    const currentUrl = new URL(`${region}/${category}`, rootUrl).href;
+
+    const { data: response } = await got(currentUrl);
+
+    const $ = cheerio.load(response);
+
+    const rootUrlRegExp = new RegExp(rootUrl.replace(/(\.|\/)/g, '\\$1'));
+
+    let items = $('ul.slide, #newssource')
+        .find('li')
+        .slice(0, limit)
+        .toArray()
+        .map((item) => {
+            item = $(item);
+
+            const a = item.find('a');
+            const image = item.find('img');
+
+            const title = item.find('h3.mediumFont').text().trim();
+
+            let link = a.prop('href');
+            link = link === '#' ? currentUrl : link;
+
+            return {
+                title,
+                link,
+                description: art(path.join(__dirname, 'templates/description.art'), {
+                    description: item.find('p.gowhite').text(),
+                    image: image.prop('src')
+                        ? {
+                              src: image.prop('src'),
+                              alt: image.prop('alt'),
+                          }
+                        : undefined,
+                }),
+                guid: rootUrlRegExp.test(item.link) ? `${link}#${title}` : link,
```

Comment:
`item.link` does not exist before returning

Reply:
👌